elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,410 @@
+     * add it to the {@link EntityProjection}.
+     *
+     * @param argument graphQL argument
+     * @param entityProjection projection that has this argument
+     */
+    private void addArgument(Argument argument, EntityProjection entityProjection) {
+        String argumentName = argument.getName();
+        Value argumentValue = argument.getValue();
+
+        if (isPaginationArgument(argumentName)) {
+            addPagination(argumentName, argumentValue, entityProjection);
+        } else if (isSortingArgument(argumentName)) {
+            addSorting(argumentValue, entityProjection);
+        } else {
+            Class<?> entityType = entityProjection.getType();
+
+            if (!entityDictionary.isValidField(entityType, argumentName)) {
+                // invalid argument name
+                throw new IllegalStateException(
+                        String.format("Unknown argument field {%s.%s}.", entityType, argument));
+            }
+
+            Attribute argumentAttribute = entityProjection.getAttributeByName(argumentName);
+            com.yahoo.elide.request.Argument elideArgument = com.yahoo.elide.request.Argument.builder()
+                    .name(argumentName)
+                    .value(argumentValue)
+                    .build();
+
+            if (argumentAttribute == null) {
+                argumentAttribute = Attribute.builder()
+                        .type(entityDictionary.getType(entityType, argumentName))
+                        .name(argumentName)
+                        .argument(elideArgument)
+                        .build();
+
+                Set<Attribute> attributes = entityProjection.getAttributes();
+                attributes.add(argumentAttribute);
+
+                entityProjection.setAttributes(attributes);
+            } else {
+                argumentAttribute.getArguments().add(elideArgument);
+            }
+        }
+    }
+
+    /**
+     * Returns whether or not a GraphQL argument name corresponding to a pagination argument.
+     *
+     * @param argumentName Name key of the GraphQL argument
+     *
+     * @return {@code true} if the name equals to {@link ModelBuilder#ARGUMENT_FIRST} or
+     * {@link ModelBuilder#ARGUMENT_AFTER}
+     */
+    private static boolean isPaginationArgument(String argumentName) {
+        return ModelBuilder.ARGUMENT_FIRST.equals(argumentName) || ModelBuilder.ARGUMENT_AFTER.equals(argumentName);
+    }
+
+    /**
+     * Creates a {@link Pagination} object from pagination GraphQL argument and attaches it to the
+     * {@link EntityProjection}.
+     *
+     * @param paginationArgument  A string that contains the key to a value of sorting spec
+     * @param paginationValue  A string that contains the value of pagination spec
+     * @param entityProjection projection that has the pagination argument
+     */
+    private void addPagination(
```

Comment:
Your logic doesn't need to add this just yet.  Elide core logic adds pagination, sorting, and filters for you.